clmm/theory/func_layer.py

```diff
@@ -388,76 +405,102 @@ def compute_tangential_shear(r_proj, mdelta, cdelta, z_cluster, z_source, cosmo,
+            * 'hernquist' - not in cluster_toolkit
 
     massdef : str, optional
         Profile mass definition, with the following supported options (letter case independent):
 
-            * `mean` (default);
-            * `critical` - not in cluster_toolkit;
-            * `virial` - not in cluster_toolkit;
+            * 'mean' (default)
+            * 'critical'
+            * 'virial'
 
     alpha_ein : float, optional
-        If `halo_profile_model=='einasto'`, set the value of the Einasto slope. Option only available
-        for the NumCosmo backend
-    z_src_model : str, optional
-        Source redshift model, with the following supported options:
-            `single_plane` (default) - all sources at one redshift (if
-            `z_source` is a float) or known individual source galaxy redshifts
-            (if `z_source` is an array and `r_proj` is a float);
+        If `halo_profile_model=='einasto'`, set the value of the Einasto slope. Option only
+        availablefor the NumCosmo backend
+    z_src_info : str, optional
+        Type of redshift information provided, it describes z_source.
+        The following supported options are:
+
+            * 'discrete' (default) : The redshift of sources is provided by `z_source`.
+              It can be individual redshifts for each source galaxy when `z_source` is an array
+              or all sources are at the same redshift when `z_source` is a float.
+
+            * 'distribution' : A redshift distribution function is provided by `z_source`.
+              `z_source` must be a one dimentional function.
+
+            * 'beta' : The averaged lensing efficiency is provided by `z_source`.
+              `z_source` must be a tuple containing
+              ( :math:`\langle \beta_s \rangle, \langle \beta_s^2 \rangle`),
+              the lensing efficiency and square of the lensing efficiency averaged over
+              the galaxy redshift distribution repectively.
+
+                .. math::
+                    \langle \beta_s \rangle = \left\langle \frac{D_{LS}}{D_S}\frac{D_\infty}
+                    {D_{L,\infty}}\right\rangle
+
+                .. math::
+                    \langle \beta_s^2 \rangle = \left\langle \left(\frac{D_{LS}}
+                    {D_S}\frac{D_\infty}{D_{L,\infty}}\right)^2 \right\rangle
+
+    beta_kwargs: None, dict
+        Extra arguments for the `compute_beta_s_mean, compute_beta_s_square_mean` functions.
+        Only used if `z_src_info='distribution'`. Possible keys are:
+
+            * 'zmin' (None, float) : Minimum redshift to be set as the source of the galaxy
+              when performing the sum. (default=None)
+            * 'zmax' (float) : Maximum redshift to be set as the source of the galaxy
+              when performing the sum. (default=10.0)
+            * 'delta_z_cut' (float) : Redshift interval to be summed with $z_cl$ to return
+              $zmin$. This feature is not used if $z_min$ is provided. (default=0.1)
```

Comment:
-->  'delta_z_cut' (float) : Redshift cut so that zmin = z_cl - delta_z_cut. delta_z_cut is ignored if z_min is already provided.